rescan.go

```diff
@@ -717,13 +722,27 @@ rescanLoop:
 					log.Warnf("Received unhandled block "+
 						"notification: %T", ntfn)
 				}
+
+			// Our retry signal has fired, so we'll attempt to
+			// refetch the filter for our next queued block.
+			case <-blockRetrySignal:
+				blockRetrySignal = nil
+				if len(retryBlocks) == 0 {
+					continue rescanLoop
+				}
+
+				ntfn := retryBlocks[0]
+				err := handleBlockConnected(ntfn, true)
```

Comment:
It feels a bit odd that we already know that it is a retry, but `handleBlockConnected` is still going to check `retryBlocks`. Maybe `handleBlockConnected` should be divided in two functions such that it doesn't go through this check if it is triggered by `blockRetrySignal`. Decompose it such that it is clear what the common logic is between the retry and non-retry case.

Reply:
Fixed.